modules/gltf/src/lib/extensions/KHR_techniques_webgl.js

```diff
@@ -0,0 +1,58 @@
+// GLTF EXTENSION: KHR_techniques_webgl
+// https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_techniques_webgl
+/* global TextDecoder */
+import GLTFScenegraph from '../gltf-scenegraph';
+import {KHR_TECHNIQUES_WEBGL} from '../gltf-constants';
+
+function resolveTechniques({programs = [], shaders = [], techniques = []}, gltfScenegraph) {
+  const textDecoder = new TextDecoder();
+
+  shaders.forEach(shader => {
+    if (Number.isFinite(shader.bufferView)) {
+      shader.code = textDecoder.decode(
+        gltfScenegraph.getTypedArrayForBufferView(shader.bufferView)
+      );
+    }
+
+    // TODO: handle URI shader
+  });
+
+  programs.forEach(program => {
+    program.fragmentShader = shaders[program.fragmentShader];
```

Comment:
Convenient for app, but could lead to duplication of shaders that could otherwise be shared? 

We probably don't care at this point, this is not a very important extensions...

For other data types, we don't resolve indices to object automatically in the loader, we offer the `GLTFScenegraph` class to do that as a postProcessing step...

Reply:
Isn't the shader cache going to detect it's the same shader?